addons/account/static/src/components/account_payment_term_form/payment_term_line_ids.js

```diff
@@ -4,23 +4,21 @@ import { registry } from "@web/core/registry";
 
 import { ListRenderer } from "@web/views/list/list_renderer";
 import { X2ManyField, x2ManyField } from "@web/views/fields/x2many/x2many_field";
-
-export class PaymentTermLineIdsRenderer extends ListRenderer {
-
-    /* override */
-    onGlobalClick(ev) {
-        // Prevent the discard of new records when clicking outside of the sheet.
-        // This is needed because the user is not forced to edit something on the newly
-        // created record. Therefore, there is no reason to remove this record when he
-        // attempt to save the form.
-        this.props.list.editedRecord = null;
-        super.onGlobalClick(ev);
-    }
-
-}
+import { useAddInlineRecord } from "@web/views/fields/relational_utils";
 
 export class PaymentTermLineIdsOne2Many extends X2ManyField {
-    static components = {...X2ManyField.components, ListRenderer: PaymentTermLineIdsRenderer}
+    static components = {...X2ManyField.components, ListRenderer: ListRenderer}
+        setup() {
+        super.setup();
+        this.addInLine = useAddInlineRecord({
+            addNew: async (...args) => {
+                await this.list.addNewRecord(...args);
+                if (this.list.editedRecord?.canBeAbandoned) {
+                    this.list.editedRecord.update({});
+                }
+            }
+        });
+    }
```

Comment:
```suggestion
    static components = {...X2ManyField.components, ListRenderer: ListRenderer}
 
    setup() {
        super.setup();
        // add a comment to explain this (need not be too long)
        this.addInLine = useAddInlineRecord({
            addNew: async (...args) => {
                const newRecord = await this.list.addNewRecord(...args);
                newRecord.update({});
            }
        });
    }
```

Reply:
@FrancoisGe could you please take a look?
The previous solution (globalClick) was not good enough, as a click inside the `listRenderer` still removed the newly created `canBeAbandoned` record

Reply:
done !

Reply:
tests are on the way

Reply:
dnoe !